source/common/network/address_impl.cc

```diff
@@ -78,7 +82,14 @@ InstanceConstSharedPtr peerAddressFromFd(int fd) {
 }
 
 int InstanceBase::flagsFromSocketType(SocketType type) const {
+#if defined(__FreeBSD__)
+  int flags = O_NONBLOCK;
+#elif defined(__APPLE__)
```

Comment:
I think we should keep it since I got freeBSD to pass all tests as well. I'd double check that it's still the case. The delta between FreeBSD support and OSX is actually pretty small